dev-practices/practices.rst

```diff
@@ -0,0 +1,286 @@
+
+  /path/to/pfs_pipe2d/bin/pfs_integration_test.sh /data/pfs
+
+The full usage information for :file:`pfs_integration_test.sh` is::
+
+    Exercise the PFS 2D pipeline code
+
+    Usage: /path/to/pfs_pipe2d/bin/pfs_integration_test.sh [-b <BRANCH>] [-r <RERUN>] [-d DIRNAME] [-c CORES] [-n] <PREFIX>
+
+        -b <BRANCH> : branch of drp_stella_data to use
+        -r <RERUN> : rerun name to use (default: 'integration')
+        -d <DIRNAME> : directory name to give data repo (default: 'INTEGRATION')
+        -c <CORES> : number of cores to use (default: 1)
+        -n : don't cleanup temporary products
+        <PREFIX> : directory under which to operate
+
+.. _pfs_pipe2d: http://github.com/Subaru-PFS/pfs_pipe2d
+
+
+Continuous Integration
+======================
+
+The `pfs_pipe2d`_ package contains scripts for performing Continuous Integration builds, 
+which compile and run the unit tests of the individual packages and runs the integration test. 
+Such builds should be run before merging any work to the ``master`` branch.
+This ensures that the ``master`` branch always works. Timing will vary according to resource availability and
+the number of cores used, but the test typically takes about half an hour to run.
+
+The test can be run in one of two ways: on the developer's own system at the command-line, or on `Travis-CI`_.
+The Travis-CI method is recommended because it is requires little effort, uses external resources and provides
+a visible demonstration to the team that the code works.
+
+.. _Travis-CI: http://travis-ci.org
+
+
+Travis use
+----------
+
+The integration test will run automatically under `Travis-CI`_ when GitHub pull requests are issued for any
+of the following repositories:
+
+- pfs_pipe2d
+- drp_stella
+- drp_stella_data
+- obs_pfs
+- datamodel
+
+After issuing a pull request, all you need to do is wait for the test to finish successfully. You'll see a
+yellow circle while the test runs. You click on the "Details" button to see the build output. If it
+finished successfully, you'll get a green circle and the statement "All checks have passed". Then you're
+clear to merge after review. If the Travis check fails, you'll get a red circle, which signals that you need
+to fix your contribution. In that case, have a look at the Travis log for clues as to what the problem might
+be.
+
+As you push new commits to the pull request, Travis will automatically trigger new integration tests. Because
+Travis is triggered on GitHub pull requests, you should ensure you have pushed your work on a common ticket
+branch to all appropriate repos before making pull requests. If you want to signal Travis `not to
+automatically test a commit`_, add the text ``[ci skip]`` to your commit message.
+
+Unfortunately, due to Travis resource limitations, only 4 MB of logs can be generated. We therefore trap the
+build and test output and display only the last 100 lines when the process completes. If this makes it
```

Comment:
The code is now much less chatty, so this is no longer necessary: all output is displayed.

Reply:
Ok. Section deleted.